### DISCUSSION TITLE
[发布] 3.6.0，用 CFW 替代 GAE，修复谷歌 IP 筛选

### DISCUSSION THREAD

**Discussion by SeaHOH** [SeaHOH/GotoX#151] [公告]
## 本次更新主要内容
- 添加 Cloudflare Workers 服务端支持，替代 GAE 作为主要远程代理。
- 调整谷歌 IP 筛选。
- 改进托盘工具（Windows）。
- 其它改进和修复若干。

## 注意事项
- 增加 Cloudflare Workers 作为主要的远程代理服务器。没有部署过 CFW 服务端的新用户可以直接使用，部署方法可查看 [wiki](https://github.com/SeaHOH/GotoX/wiki/简易部署教程：CloudFlare-Workers)。
- 由于谷歌政策调整，未绑定信用卡的 cloud 账户 5 月 31 日后将无法使用其免费 GAE 服务，因此不再继续改进 GAE 服务端。建议老用户转为使用 CFW 代理，否则请自行部署。
- 由于谷歌有调整了服务器，建议老用户不要保留旧的 data 文件夹，**如果需要保留**，请设置 [picker/strict]，不然无法正常直连谷歌。GWS IP 的筛选线程不要设置得过高，请保持在个位数，否则可能会触发屏蔽（具体是谷歌还是 GFW，我现在已经没精力去分辨）。

## CFW 服务端的一些问题
- HTTP 出口请求不支持非标准端口（**存疑，请求更多测试样本**）。
- 单个 Worker 使用内存上限为 128MB，所以不要同时下载大量比较大的文件，多线程下载也要注意线程数量，上传文件大小也可能（**尚未测试**）受此限制。
- CFW 代理使用 CloudFlare 节点连接，默认解析主站得到 CDN 节点，如果速度和质量不满意，可尝试自定义其它节点（详见设置文件）。

**Comment by axa888** [comment]
最新版本出现这种情况
Thread 0x0000041c (most recent call first):
  File "F:\Proxy\GotoX\python\site-packages\gevent-20.5.0-cp38-cp38-win_amd64.eg
g\gevent\_threading.py", line 60 in wait
  File "F:\Proxy\GotoX\python\site-packages\gevent-20.5.0-cp38-cp38-win_amd64.eg
g\gevent\_threading.py", line 146 in get
  File "F:\Proxy\GotoX\python\site-packages\gevent-20.5.0-cp38-cp38-win_amd64.eg
g\gevent\threadpool.py", line 155 in run

Current thread 0x00000c98 (most recent call first):
  File "F:\Proxy\GotoX\python\site-packages\gevent-20.5.0-cp38-cp38-win_amd64.eg
g\gevent\_ffi\loop.py", line 264 in python_stop
  File "F:\Proxy\GotoX\python\site-packages\gevent-20.5.0-cp38-cp38-win_amd64.eg
g\gevent\libuv\loop.py", line 498 in __run_queued_callbacks
  File "F:\Proxy\GotoX\python\site-packages\gevent-20.5.0-cp38-cp38-win_amd64.eg
g\gevent\libuv\loop.py", line 218 in _run_callbacks
  File "F:\Proxy\GotoX\python\site-packages\gevent-20.5.0-cp38-cp38-win_amd64.eg
g\gevent\_ffi\loop.py", line 300 in python_prepare_callback
  File "F:\Proxy\GotoX\python\site-packages\gevent-20.5.0-cp38-cp38-win_amd64.eg
g\gevent\libuv\loop.py", line 542 in run
  File "F:\Proxy\GotoX\python\site-packages\gevent-20.5.0-cp38-cp38-win_amd64.eg
g\gevent\hub.py", line 618 in run

**Comment by SeaHOH** [comment]
@axa888 试试在托盘工具中切换 gevent 设置，位置在菜单第一个项目中。
如果换用其它设置没有效果，你的旧版本还正常，请下载 GotoX-3.6.0-py3-none-any.zip 覆盖旧版。

**Comment by axa888** [comment]
把3.6.0a中的python整个文件夹替换到3.6.0中就正常了,切换gevent设置也可以。

**Comment by lxf1009** [comment]
@SeaHOH 和2楼一样问题，试了你说的所有办法，都无效。

**Comment by SeaHOH** [comment]
@lxf1009 第二个方法怎么都行吧，除非你的旧版运行也出错。直接替换 3.6.0a 的 python 文件夹也行，一回事。

顺便问问你们的操作系统是 Win10？

**Comment by standritsu** [comment]
我的走8086出现这个错误：

Exception happened during processing of request from ('::1', 60909, 0, 0)
Traceback (most recent call last):
  File "D:\Program Files\GotoX-win_amd64\local\ProxyServer.py", line 125, in process_request_thread
    self.finish_request(request, client_address)
  File "D:\obj\windows-release\37amd64_Release\msi_python\zip_amd64\socketserver.py", line 360, in finish_request
  File "D:\Program Files\GotoX-win_amd64\local\ProxyHandler.py", line 129, in __init__
    self.handle()
  File "D:\obj\windows-release\37amd64_Release\msi_python\zip_amd64\server.py", line 426, in handle
  File "D:\Program Files\GotoX-win_amd64\local\ProxyHandler.py", line 186, in handle_one_request
    self.do_CONNECT()
  File "D:\Program Files\GotoX-win_amd64\local\ProxyHandler.py", line 1484, in do_CONNECT
    self.do_action()
  File "D:\Program Files\GotoX-win_amd64\local\ProxyHandler.py", line 232, in do_action
    getattr(self, self.action)()
  File "D:\Program Files\GotoX-win_amd64\local\ProxyHandler.py", line 1138, in do_FAKECERT
    self.handle()
  File "D:\obj\windows-release\37amd64_Release\msi_python\zip_amd64\server.py", line 426, in handle
  File "D:\Program Files\GotoX-win_amd64\local\ProxyHandler.py", line 188, in handle_one_request
    self.do_METHOD()
  File "D:\Program Files\GotoX-win_amd64\local\ProxyHandler.py", line 1491, in do_METHOD
    self.do_action()
  File "D:\Program Files\GotoX-win_amd64\local\ProxyHandler.py", line 232, in do_action
    getattr(self, self.action)()
  File "D:\Program Files\GotoX-win_amd64\local\ProxyHandler.py", line 647, in do_CFW
    if self.target and '@follow' in self.target:
AttributeError: 'ACTProxyHandler' object has no attribute 'target'

**Comment by SeaHOH** [comment]
> AttributeError: 'ACTProxyHandler' object has no attribute 'target'

稍后修复，请先使用自动端口。

**Comment by lxf1009** [comment]
@SeaHOH 替换也不行，旧版一直运行良好。我系统是WIN7。

**Comment by yccsword** [comment]
新版CFW是否是不支持代理github,默认配置仅配置cfworker,代理访问github提示如下:
L4:51223->140.82.112.26 "FWD CONNECT live.github.com:443 HTTP/1.1" - -
17:08:58 W 192.30.253.112 _create_ssl_connection 'github.com' 返回 timeout('timed out')，重试
17:08:59 W 192.30.253.112 create_ssl_connection 'https://github.com/' 失败：timeout('timed out')
17:08:59 I L4:65508->[2607:f8b0:4007:808::2000] "DIRECT POST https://clients4.google.com/chrome-sync/command/?client=Google+Chrome&client_id=aW0b/9NgCssminstlPQLWA%3D%3D HTTP/1.1" 200 -
17:09:00 W 192.30.253.112 _create_ssl_connection 'github.com' 返回 timeout('timed out')，重试
17:09:01 W 192.30.253.112 create_ssl_connection 'https://github.com/' 失败：timeout('timed out')
17:09:02 W 192.30.253.112 _create_ssl_connection 'github.com' 返回 timeout('timed out')，重试
17:09:03 W 192.30.253.112 create_ssl_connection 'https://github.com/' 失败：timeout('timed out')
17:09:04 W 192.30.253.112 _create_ssl_connection 'github.com' 返回 timeout('timed out')，重试
17:09:05 W 192.30.253.112 create_ssl_connection 'https://github.com/' 失败：timeout('timed out')
17:09:06 W 192.30.253.112 _create_ssl_connection 'github.com' 返回 timeout('timed out')，重试

**Comment by standritsu** [comment]
自动代理访问www.top1health.com的时候出现下面的报错：
Exception happened during processing of request from ('::1', 49375, 0, 0)
Traceback (most recent call last):
  File "D:\Program Files\GotoX-win_amd64\local\ProxyServer.py", line 125, in process_request_thread
    self.finish_request(request, client_address)
  File "D:\obj\windows-release\37amd64_Release\msi_python\zip_amd64\socketserver.py", line 360, in finish_request
  File "D:\Program Files\GotoX-win_amd64\local\ProxyHandler.py", line 129, in __init__
    self.handle()
  File "D:\obj\windows-release\37amd64_Release\msi_python\zip_amd64\server.py", line 426, in handle
  File "D:\Program Files\GotoX-win_amd64\local\ProxyHandler.py", line 186, in handle_one_request
    self.do_CONNECT()
  File "D:\Program Files\GotoX-win_amd64\local\ProxyHandler.py", line 288, in do_CONNECT
    self.do_action()
  File "D:\Program Files\GotoX-win_amd64\local\ProxyHandler.py", line 232, in do_action
    getattr(self, self.action)()
  File "D:\Program Files\GotoX-win_amd64\local\ProxyHandler.py", line 1138, in do_FAKECERT
    self.handle()
  File "D:\obj\windows-release\37amd64_Release\msi_python\zip_amd64\server.py", line 426, in handle
  File "D:\Program Files\GotoX-win_amd64\local\ProxyHandler.py", line 188, in handle_one_request
    self.do_METHOD()
  File "D:\Program Files\GotoX-win_amd64\local\ProxyHandler.py", line 314, in do_METHOD
    self.do_action()
  File "D:\Program Files\GotoX-win_amd64\local\ProxyHandler.py", line 232, in do_action
    getattr(self, self.action)()
  File "D:\Program Files\GotoX-win_amd64\local\ProxyHandler.py", line 604, in do_DIRECT
    raise e
  File "D:\Program Files\GotoX-win_amd64\local\ProxyHandler.py", line 566, in do_DIRECT
    return self.go_GAE()
AttributeError: 'AutoProxyHandler' object has no attribute 'go_GAE'

手动添加自动代理规则走CFW没问题

**Comment by shpherd** [comment]
在简易部署教程：CloudFlare Workers 步骤7 复制 GotoX/remote/cloudflare_workers/index.js 内容到左边的代码框，deploy后
uncaught syntaxerror : invalid or unexpected token at line1 。
截图无法上传

**Comment by SeaHOH** [comment]
@lxf1009 你把新版的 python 文件夹删掉，复制旧版的进去。

@yccsword 你没有使用 CFW 连接 github.com，需要设置自动代理规则，可查看配置文件顶部说明。

> AttributeError: 'AutoProxyHandler' object has no attribute 'go_GAE'

稍后修复。

@shpherd 可能你的编辑器编码不符，尝试用 notepad2 或 notepad++ 等打开复制。

**Comment by hg123** [comment]
Traceback (most recent call last):
  File "C:\Users\Administrator\Desktop\GotoX-3.6.0-cp38\start.py", line 14, in <module>
    proxy.main()
  File "C:\Users\Administrator\Desktop\GotoX-3.6.0-cp38\local\proxy.py", line 294, in main
    network_test()
  File "C:\Users\Administrator\Desktop\GotoX-3.6.0-cp38\local\common\decorator.py", line 35, in newfunc
    result = func(*args, **kwargs)
  File "C:\Users\Administrator\Desktop\GotoX-3.6.0-cp38\local\ProxyServer.py", line 25, in network_test
    if not is_active(type):
  File "C:\Users\Administrator\Desktop\GotoX-3.6.0-cp38\local\common\internet_active.py", line 284, in is_active
    start_new_thread(_is_active, (4, qobj, keep_on))
RuntimeError: can't start new thread

我这里会出现这种错误

**Comment by SeaHOH** [comment]
> RuntimeError: can't start new thread

如果你在禁用 gevent 的情况下设置了非常大的筛选线程，可能会导致这个错误。

否则，试一下重启程序或系统，感觉像是内存错误。

如果还是不行，试一下其它 gevent 设置。

**Comment by hg123** [comment]
我都是默认设置 
上一个版本没有出现这个问题 
一般是在长时间开启后会这样

**Comment by hg123** [comment]
你说到gevent设置 我去对比了一下二个版本的设置 上一版是默认 这一版是禁用 我把这一版改回默认看看

**Comment by caofengzi** [comment]
以前版本俺的win7可以啟動使用，昨天下載新版無法啟動了，進程裡面也沒有gotox進程，不知道什麼原因。

**Comment by SeaHOH** [comment]
> 昨天下載新版無法啟動了，進程裡面也沒有gotox進程

是否下载完全？错误信息？

**Comment by shpherd** [comment]
我的也是，运行GotoX-3.6.1-cp38-win_amd64下载程序没东西，没提示。之前试验版本正常。

**Comment by SeaHOH** [comment]
把以下 cmd 命令保存到 GotoX.vbs 所在目录，如 GotoX.bat。运行后会留下命令行窗口，可以从这里复制错误信息。
```bat
set PYTHONPATH=
set PYTHONHOME=
cd python
python ..\launcher\start.py
pause
```

**Comment by shpherd** [comment]
E:\chrome plugins\GotoX-3.6.1-cp38-win_amd64>set PYTHONPATH=

E:\chrome plugins\GotoX-3.6.1-cp38-win_amd64>set PYTHONHOME=

E:\chrome plugins\GotoX-3.6.1-cp38-win_amd64>cd python

E:\chrome plugins\GotoX-3.6.1-cp38-win_amd64\python>python ..\launcher\start.py

Traceback (most recent call last):
  File "..\launcher\start.py", line 19, in <module>
    import win_tray
  File "E:\chrome plugins\GotoX-3.6.1-cp38-win_amd64\launcher\win_tray.py", line
 12, in <module>
    from common import (
  File "E:\chrome plugins\GotoX-3.6.1-cp38-win_amd64\launcher\common.py", line 5
, in <module>
    import ssl
  File "<frozen zipimport>", line 259, in load_module
  File "ssl.py", line 98, in <module>
ImportError: DLL load failed while importing _ssl: 参数错误。

E:\chrome plugins\GotoX-3.6.1-cp38-win_amd64\python>pause
请按任意键继续. . .

**Comment by SeaHOH** [comment]
> File "", line 259, in load_module

这里是什么？如果你没自己修改过文件，替换 `python ..\launcher\start.py` 为 `python -E ..\launcher\start.py` 再运行一次。

**Comment by caofengzi** [comment]
> > 昨天下載新版無法啟動了，進程裡面也沒有gotox進程
> 
> 是否下载完全？错误信息？

俺是32位win7，今天重新下載還是不能啟動，初次點擊gotox.vbs啟動命令行窗口會閃一下，然後就沒有了，進程也沒有gotox進程，但是托盤會有啟動圖標，剛才按照你上面把cmd文件保存為bat在啟動，這次連托盤圖標都沒有了。啟動不了所以沒有錯誤信息。

**Comment by SeaHOH** [comment]
> 啟動不了所以沒有錯誤信息。

不可能什么都没有，至少会有像下面这样的命令回显。

> E:\chrome plugins\GotoX-3.6.1-cp38-win_amd64>set PYTHONPATH=
> 
> E:\chrome plugins\GotoX-3.6.1-cp38-win_amd64>set PYTHONHOME=
> 
> E:\chrome plugins\GotoX-3.6.1-cp38-win_amd64>cd python
> 
> E:\chrome plugins\GotoX-3.6.1-cp38-win_amd64\python>python ..\launcher\start.py

**Comment by shpherd** [comment]
E:\chrome plugins\GotoX-3.6.1-cp38-win_amd64>set PYTHONPATH=

E:\chrome plugins\GotoX-3.6.1-cp38-win_amd64>set PYTHONHOME=

E:\chrome plugins\GotoX-3.6.1-cp38-win_amd64>cd python

E:\chrome plugins\GotoX-3.6.1-cp38-win_amd64\python>python -E ..\launcher\start.
py
Traceback (most recent call last):
  File "..\launcher\start.py", line 19, in <module>
    import win_tray
  File "E:\chrome plugins\GotoX-3.6.1-cp38-win_amd64\launcher\win_tray.py", line
 12, in <module>
    from common import (
  File "E:\chrome plugins\GotoX-3.6.1-cp38-win_amd64\launcher\common.py", line 5
, in <module>
    import ssl
  File "<frozen zipimport>", line 259, in load_module
  File "ssl.py", line 98, in <module>
ImportError: DLL load failed while importing _ssl: 参数错误。

E:\chrome plugins\GotoX-3.6.1-cp38-win_amd64\python>pause
请按任意键继续. . .

**Comment by SeaHOH** [comment]
> File "", line 259, in load_module

请告诉我这是什么，可发邮件。

**Comment by shpherd** [comment]
GotoX-3.6.1-cp38-win_amd64.7z  通过http://gitd.cc/  下载。
复制内容少了一行。还是line 259
 File "<frozen zipimport>", line 259, in load_module
file frozen zipimport , line 259,  in load_module

**Comment by caofengzi** [comment]
> > 啟動不了所以沒有錯誤信息。
> 
> 不可能什么都没有，至少会有像下面这样的命令回显。
> 
> > E:\chrome plugins\GotoX-3.6.1-cp38-win_amd64>set PYTHONPATH=
> > E:\chrome plugins\GotoX-3.6.1-cp38-win_amd64>set PYTHONHOME=
> > E:\chrome plugins\GotoX-3.6.1-cp38-win_amd64>cd python
> > E:\chrome plugins\GotoX-3.6.1-cp38-win_amd64\python>python ..\launcher\start.py

点击gotox.vbs启动时命令行一闪就没了，啥信息也看不到的。

**Comment by SeaHOH** [comment]
> 点击gotox.vbs启动时命令行一闪就没了，啥信息也看不到的。

运行上面保存的 bat 文件，贴出显示的**所有**信息。

**Comment by SeaHOH** [comment]
> GotoX-3.6.1-cp38-win_amd64.7z 通过http://gitd.cc/ 下载。

是否验证过哈希？可能是下载不完全或解压缩错误。
GotoX-3.6.1-cp38-win_amd64\python\DLLs 文件夹下面是否有 _ssl.pyd 文件？

**Comment by caofengzi** [comment]
> > 点击gotox.vbs启动时命令行一闪就没了，啥信息也看不到的。
> 
> 运行上面保存的 bat 文件，贴出显示的**所有**信息。

奇怪了昨天不能运行bat，今天可以了，这是显示信息。
C:\Users\Administrator\Pictures\GotoX-3.6.1-cp38-win32>set PYTHONPATH=

C:\Users\Administrator\Pictures\GotoX-3.6.1-cp38-win32>set PYTHONHOME=

C:\Users\Administrator\Pictures\GotoX-3.6.1-cp38-win32>cd python

C:\Users\Administrator\Pictures\GotoX-3.6.1-cp38-win32\python>python ..\launcher
\start.py
Traceback (most recent call last):
  File "..\launcher\start.py", line 19, in <module>
    import win_tray
  File "C:\Users\Administrator\Pictures\GotoX-3.6.1-cp38-win32\launcher\win_tray
.py", line 12, in <module>
    from common import (
  File "C:\Users\Administrator\Pictures\GotoX-3.6.1-cp38-win32\launcher\common.p
y", line 5, in <module>
    import ssl
  File "<frozen zipimport>", line 259, in load_module
  File "ssl.py", line 98, in <module>
ImportError: DLL load failed while importing _ssl: 参数错误。

C:\Users\Administrator\Pictures\GotoX-3.6.1-cp38-win32\python>pause

**Comment by SeaHOH** [comment]
一模一样的错误，你也是在其它地方下载的吗？如果是，参见上面 https://github.com/SeaHOH/GotoX/issues/151#issuecomment-678630532 。

**Comment by caofengzi** [comment]
> 一模一样的错误，你也是在其它地方下载的吗？如果是，参见上面 [#151 (comment)](https://github.com/SeaHOH/GotoX/issues/151#issuecomment-678630532) 。

俺是在这里下载的，而且对比过哈希值了没有错误。

**Comment by SeaHOH** [comment]
系统里面是否还安装有其它 python？
运行 GotoX-3.6.1-cp38\python 文件夹下面的 python.exe，输入以下代码，把输出贴一下。
```py
import sys
sys.path
```

**Comment by caofengzi** [comment]
> 系统里面是否还安装有其它 python？
> 运行 GotoX-3.6.1-cp38\python 文件夹下面的 python.exe，输入以下代码，把输出贴一下。
> 
> ```python
> import sys
> sys.path
> ```

没有安装Python，一直都是用gotox自带的。

**Comment by caofengzi** [comment]
> 系统里面是否还安装有其它 python？
> 运行 GotoX-3.6.1-cp38\python 文件夹下面的 python.exe，输入以下代码，把输出贴一下。
> 
> ```python
> import sys
> sys.path
> ```

这是按照上面得到的信息。
Python 3.8.4（tags/u3.8.4：dfa645a，Jul 13 2020，16：30：28）[MSC U.1926 32 bit（In tel）]
on win32
Type help"，"copyr ight"，”credits"or”license"for more information.
>>> import sys
>>> sys.path
[，'C:\Users\Administratortictures\\GotoX-3.6.1-cp38-win32\\python python
38.zip i C：\Users\\AdministratorPicturesWGotoX-3.6.1-cp38-win32\\python\\DL Ls'，”C:\\Users\AdministratortPictures\Gotox-3 6.1-cp38-win32\\python，'C:W Users\\Administrator\Pic tures\lGotoX-3.6.1-cp38-win32Wpythonisi te-packages'，
'C：\Users\AdministratortPictures\GotoX-3.6.1-cp38-win32W\python\isi te-packages\\cffi-1.14.0-cp38-cp38-win32.egg'，'C：UsersMAdministratorPictures\lGotox
-3.6.1-cp38-win32V\pythonisi te-packages\\cryptography-2.9.2-cp38-cp38-win32.eg9
'C：Users\\AdministratorAictures\\GotoX-3 6.1-cp38-win32\\python\\si te-pac kages\\dns1ib-0.9.14.egg’，'C：\Users\\Administratorlictures\lGotoX-3.6.1-cp38
-win32\python\\si te-packages\\gevent-20.6.2-cp38-cp38-win32.egg'，'C：UsersAAdministrator\\Pictures\lGotoX-3.6.1-cp38-win32\python\\si te-packages\igreenlet-0.4.16-cp38-cp38-win32.egg'，'C：\Users\\AdministratorPictures\lGotoX-3.6.1-cp
38-win32\python\\site-packageshelpers-win32.e9g'，'C：\Users\\Administrator Pictures\\GotoX-3.6.1-cp38-win32\\python\\si te-packages\\pkg_resources-49.2.0-py
3-none-any.e9g'，'C：\Users\Administrator\ictures\\GotoX-3.6.1-cp38-win32\\py thon\\si te-packages\\pyOpenSSL-19.1 0-py2.py3-none-any，egg'，'C:\Users\\Adminis trator\\Pictures\\GotoX-3.6.1-cp38-win32Wpython\\si te一packages\PySocks-1.7.1-py3-none-any.egg’，'C：\Users\AdministratorlPictures\lGotoX-3.6.1-cp38-win32W\p ython\isi te-packages\\six-1.15.0-py2.py3-none一any.egg'，c：Users\\Administrato r\\Pictures\\GotoX-3.6.1-cp38-win32\\python\\si te-packages\\zope.event-4.4-py2.p y3-none-any.egg'，'C：\Users\AdministratorlPictureslGotoX-3.6.1-cp38-win32V\p ython\\si te-packages\\zope.inter face-5.1.0-cp38-cp38-win32.egg']

**Comment by SeaHOH** [comment]
为什么你贴的内容这么奇怪，居然还有汉字？
如果忽略以上问题，似乎是正常的。

看看 python\DLLs 文件夹下面是否有 _ssl.pyd、libcrypto-1_1.dll、libssl-1_1.dll 这三个文件？

**Comment by caofengzi** [comment]
> 为什么你贴的内容这么奇怪，居然还有汉字？
> 如果忽略以上问题，似乎是正常的。
> 
> 看看 python\DLLs 文件夹下面是否有 _ssl.pyd、libcrypto-1_1.dll、libssl-1_1.dll 这三个文件？

今天试了头3个版本，只有3.6.0a能够启动运行，俺用3.6.0a的Python 的文件替换其它两个版本里面的Python文件还是不能启动，用Python 3.8文件替换3.6.0a的Python文件后3.6.0a也不能启动了。

**Comment by SeaHOH** [comment]
> 今天试了头3个版本，只有3.6.0a能够启动运行，俺用3.6.0a的Python 的文件替换其它两个版本里面的Python文件还是不能启动

你是怎么替换的？必须替换整个 python 文件夹，而不是只替换其中部分文件。

**Comment by caofengzi** [comment]
> > 今天试了头3个版本，只有3.6.0a能够启动运行，俺用3.6.0a的Python 的文件替换其它两个版本里面的Python文件还是不能启动
> 
> 你是怎么替换的？必须替换整个 python 文件夹，而不是只替换其中部分文件。

俺就是替换Python的完整文件夹的，另外自动代理也不能访问GitHub，还有就是需要走代理访问的网站都要手动添加规则吗？YouTube添加规则也不能访问。

**Comment by SeaHOH** [comment]
换用安装版 python 试一试。
1. 下载安装官方 python 3.7.x - 3.8.x 任一版本，安装时启用环境变量选项。
1. 运行命令 `python -m ensurepip` 安装 pip 包管理工具。
1. 克隆 GotoX 仓库到本地，或者下载 GotoX 源码并解压缩到单独文件夹，然后进入 GotoX 文件夹。
1. 运行命令 `pip3 install requirements.txt` 安装依赖包
1. 双击 GotoX.vbs 运行。

**Comment by SeaHOH** [comment]
> 自动代理也不能访问GitHub，还有就是需要走代理访问的网站都要手动添加规则吗？

部分网站可以自动使用 CFW 规则，根据网络状况会有 10 到 60 秒左右的等待时间，就像以前的自动 GAE 规则那样。
建议把常用站点明确添加到 CFW 规则中。

> YouTube添加规则也不能访问。

你是怎么添加的？

**Comment by caofengzi** [comment]
> > 自动代理也不能访问GitHub，还有就是需要走代理访问的网站都要手动添加规则吗？
> 
> 部分网站可以自动使用 CFW 规则，根据网络状况会有 10 到 60 秒左右的等待时间，就像以前的自动 GAE 规则那样。
> 建议把常用站点明确添加到 CFW 规则中。
> 
> > YouTube添加规则也不能访问。
> 
> 你是怎么添加的？

与你的例子一样。
YouTube.com$

**Comment by SeaHOH** [comment]
> 与你的例子一样。
> YouTube.com$

你是添加在文件末尾？这是无效的，因为前面还有个优先级更高的直连规则。
请仔细阅读文件开头的说明，注意规则起效的优先顺序。

另外，使用安装版 python 运行 GotoX 是否正常？

**Comment by caofengzi** [comment]
> > 与你的例子一样。
> > YouTube.com$
> 
> 你是添加在文件末尾？这是无效的，因为前面还有个优先级更高的直连规则。
> 请仔细阅读文件开头的说明，注意规则起效的优先顺序。
> 
> 另外，使用安装版 python 运行 GotoX 是否正常？
前面末尾都不行，不想折腾了，经常访问的站点能用就行了。

**Comment by caofengzi** [comment]
有些网页加载不完整，比如：论坛插件图片等等，是服务端的问题吗？

**Comment by SeaHOH** [comment]
> 有些网页加载不完整，比如：论坛插件图片等等，是服务端的问题吗？

不知道，估计不是，你给个链接我看看。
另外，你使用的代码是否已更新到最新的主分支？如果没有，先下载覆盖更新。

**Comment by caofengzi** [comment]
> > 有些网页加载不完整，比如：论坛插件图片等等，是服务端的问题吗？
> 
> 不知道，估计不是，你给个链接我看看。
> 另外，你使用的代码是否已更新到最新的主分支？如果没有，先下载覆盖更新。

https://program-think.blogspot.com/
3.6.1不是最新版本吗？启动不了。

**Comment by SeaHOH** [comment]
> 3.6.1不是最新版本吗？启动不了。

从这里下载 https://github.com/SeaHOH/GotoX/archive/master.zip

**Comment by caofengzi** [comment]
> > 3.6.1不是最新版本吗？启动不了。
> 
> 从这里下载 https://github.com/SeaHOH/GotoX/archive/master.zip

不是服务端问题是版本原因，下载了没有打包的文件后，俺把3.6.0a的Python和launcher文件复制过去可以启动了，启动后显示版本3.6.1，谢谢。

**Comment by caofengzi** [comment]
通过这几天折腾发现俺这里3.6.1无法启动的原因。1，Python版本原因，2，launcher文件夹里面的两个py文件原因，文件名分别是：common.py，win_tray.py，3，DLLs文件原因，替换3.6.0a的三个文件就可以启动运行。

**Comment by caofengzi** [comment]
![20200827100009](https://user-images.githubusercontent.com/13650252/91375932-8dbf3e00-e7e1-11ea-99ad-798fa2663f66.jpeg)

**Comment by SeaHOH** [comment]
你试一下解压缩 3.6.1 后不要进行任何修改，直接运行。

**Comment by caofengzi** [comment]
> 你试一下解压缩 3.6.1 后不要进行任何修改，直接运行。

已经试过很多次了，奇怪了只要有3.8版本的文件就不能启动。
